### PR TITLE
On demand Case changes

### PR DESCRIPTION
**Pull request by sogoresman**
**Description:** default to falsy when these values do know exist in the relevant block

Bug in secret-splunk.yaml - Forces usage of client key and cert even when forwarding logs to a http event endpoint for a saas instance. Doubly problematic as we inject secrets into these pods 

**Link to Splunk idea:** <Link to Splunk idea, see https://ideas.splunk.com>

**Testing:** working with professional services who are consulting internal dev to test

### CODE REVIEW

**Comment by sogoresman** [comment]
recheck

**Comment by jvoravong** [comment]
@sogoresman please make sure to sign the CLA and create a changelog entry for this PR. You may want to check out [CONTRIBUTING.md](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/CONTRIBUTING.md).

File: helm-charts/splunk-otel-collector/templates/secret-splunk.yaml [hunk 1]
```diff
@@ -17,13 +17,13 @@ data:
   {{- if (eq (include "splunk-otel-collector.splunkPlatformEnabled" .) "true") }}
   splunk_platform_hec_token: {{ .Values.splunkPlatform.token | b64enc }}
   {{- end }}
-  {{- with .Values.splunkPlatform.clientCert }}
+  {{- with .Values.splunkPlatform.clientCert | default nil }}
```
Comment (sbylica-splunk) [inline helm-charts/splunk-otel-collector/templates/secret-splunk.yaml:20]: Should be `| default ""` as it is a default value in values.yaml for all of these.

**Comment by sogoresman** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by sogoresman** [comment]
recheck

File: helm-charts/splunk-otel-collector/templates/secret-splunk.yaml (hunk 1 above)
Comment (sogoresman) [reply helm-charts/splunk-otel-collector/templates/secret-splunk.yaml]: Thanks for the addition, fixed, packaged locally and test in a call with on demand case

Comment (sbylica-splunk) [reply helm-charts/splunk-otel-collector/templates/secret-splunk.yaml]: Ok, looks good. Still missing the changelog though

**Pull request closed without merging**